api/app.py

```diff
@@ -84,6 +85,9 @@ def create_flask_app_with_configs() -> Flask:
     dify_app = DifyApp(__name__)
     dify_app.config.from_mapping(dify_config.model_dump())
 
+    if dify_app.config.get("PROXY_FIX_MIDDLEWARE_ENABLED"):
+        dify_app.wsgi_app = ProxyFix(dify_app.wsgi_app)
```

Comment:
Never decorate the flask app instance inside the `api/app.py`.
Try to create a dedicate extension like `ext_compress.py`.

Reply:
Changed to implement this as an new extension `ext_proxy_fix`.